docs/external_hms.md

```diff
@@ -0,0 +1,38 @@
+# External HMS Integration
+### TL;DR
+The UCX toolkit by default relies on the internal workspace HMS as a source for tables and views.
+<br/>Many DB users utilize an external HMS instead of the Workspace HMS provided by DB.
+<br/>A popular external HMS is Amazon Glue.
+<br/>This document describes the considerations for UCX integration with external HMS.
+
+### Current Considerations
+- Integration with external HMS is set up on individual clusters.
+- Theoretically we can integrate separate clusters in a workspace with different HMS repositories.
+- In reality most customers use a single (internal or external) HMS within a workspace.
+- When migrating from an external HMS we have to consider that it is used by more than one workspace.
+- Integration with external HMS has to be set on all DB Warehouses together.
+- HMS connectivity is set, usually, on cluster policy.
+- Typically external HMS setup relies on:
+  - Spark Config
+  - Instance Profiles
+  - Init scripts
+
+### Design Decisions
+- We should set up a single HMS for UCX
+- We should suggest copying the setup from an existing Cluster/Cluster policy
+- We shouldn't override the set up for the DB Warehouses (that may break functionality)
+- We should allow overriding cluster settings and instance profile setting to accommodate novel settings.
+
+### Challenges
+- We cannot support multiple HMS
+- Using an external HMS to persist UCX tables will break functionality for a second workspace using UCX
+- We should consider using a pattern similar to our integration testing to rename the target database to allow persisting from multiple workspaces. For example WS1 --> UCX_ABC, WS2 --> UCX_DEF.
```

Comment:
If multiple workspaces are sharing the same Ext HMS, we should do table crawl just once but should do multiple grants crawl per each workspace

Reply:
changing the code to only crawl the external HMS once is a premature optimization.  This operation only takes a few minutes right now.